tools/filterdiff.py

```diff
@@ -0,0 +1,150 @@
+# /usr/bin/env python3
+import sys
+import os
+import argparse
+import tempfile
+import json
+import subprocess
+from typing import List
+
+
+def filter_fn(imi, unwanted_tmpfiled_d: List):
+    """
+    Filter specific fields in the image-info that can't be compared
+    together.
+    """
+    def tmpfiles_d(imi, to_remove: List):
+        """
+        Remove certain files from the tpmfiles_d that have a tendency of having
+        a non fixed content.
+        """
+        if imi:
+            tmpfilesd = imi.get("tmpfiles.d")
+            if tmpfilesd:
+                for _, v in tmpfilesd.items():
+                    for trm in to_remove:
+                        if trm in v:
+                            del v[trm]
+        return imi
+
+    def lvm2(imi):
+        """
+        LVM2 partitions have a UUID that is not fixed. Replace the value
+        upon comparison time.
+        """
+        if imi:
+            partitions = imi.get("partitions")
+            if partitions:
+                for partition in partitions:
+                    if partition.get("fstype") == "LVM2_member":
+                        partition["uuid"] = "2022-07-01-fixed-uuid"
+        return imi
+
+    def iso(imi):
+        """
+        For isos, the partition UUID is the date of the build. Replace
+        that with a fixed one for the comparison.
+        """
+        if imi:
+            if ("image-format" in imi and "type" in imi["image-format"] and
+                    imi["image-format"]["type"] == "raw"):
+                if "partitions" in imi:
+                    for partition in imi["partitions"]:
+                        if "fstype" in partition:
+                            if partition["fstype"] == "iso9660":
+                                partition["uuid"] = "2022-07-01-fixed-uuid"
+            return imi
+    return iso(lvm2(tmpfiles_d(imi, unwanted_tmpfiled_d)))
```

Comment:
nitpick: this will get hard to read once more functions accept additional arguments. Calling filtering functions in a loop could be easier to read, but I do not insist on it.

Reply:
Can do, since I'm changing that to a class in any case. Maybe not a loop, I'll see with what I come up to. Thanks for the review!